homeassistant/components/websocket_api/commands.py

```diff
@@ -202,3 +204,43 @@ def handle_ping(hass, connection, msg):
     Async friendly.
     """
     connection.send_message(pong_message(msg["id"]))
+
+
+@callback
+@decorators.websocket_command(
+    {
+        vol.Required("type"): "render_template",
+        vol.Required("template"): cv.template,
+        vol.Optional("entity_ids"): cv.entity_ids,
+        vol.Optional("variables"): dict,
+    }
+)
+def handle_render_template(hass, connection, msg):
+    """Handle render_template command.
+
+    Async friendly.
+    """
+    template = msg["template"]
+    template.hass = hass
+
+    variables = msg.get("variables")
+
+    entity_ids = msg.get("entity_ids")
+    if entity_ids is None:
+        entity_ids = template.extract_entities(variables)
+
+    @callback
+    def state_listener(*_):
+        connection.send_message(
+            messages.event_message(
+                msg["id"], {"result": template.async_render(variables)}
+            )
+        )
+
+    if entity_ids and entity_ids != MATCH_ALL:
+        connection.subscriptions[msg["id"]] = async_track_state_change(
```

Comment:
Add an `else` that creates an no-op unsub function. The frontend doesn't know if a subscription was set up, so will always call unsub, which would then cause an error.